platform/macos/src/MGLMapView.mm

```diff
@@ -126,6 +127,46 @@
     return { p1[0], p1[1], p2[0], p2[1] };
 }
 
+std::vector<mbgl::LatLng> coordinatesForShape(MGLShape *shape) {
+    std::vector<mbgl::LatLng> latLngs;
+    if ([shape isKindOfClass:[MGLMultiPoint class]]) {
+        MGLMultiPoint *multiPoint = (MGLMultiPoint *)shape;
+        latLngs.reserve(multiPoint.pointCount);
+        CLLocationCoordinate2D *coordinates = multiPoint.coordinates;
+        for(int i=0; i < [multiPoint pointCount]; i++) {
+            latLngs.push_back({coordinates[i].latitude, coordinates[i].longitude});
+        }
+    } else if ([shape isKindOfClass:[MGLPointCollection class]]) {
+        MGLPointCollection *pointCollection = (MGLPointCollection *)shape;
+        latLngs.reserve(pointCollection.pointCount);
+        CLLocationCoordinate2D *coordinates = pointCollection.coordinates;
+        for(int i=0; i < [pointCollection pointCount]; i++) {
+            latLngs.push_back({coordinates[i].latitude, coordinates[i].longitude});
+        }
+    } else if ([shape isKindOfClass:[MGLMultiPolygon class]]) {
+        MGLMultiPolygon *shapeCollection = (MGLMultiPolygon *)shape;
+        for(MGLPolygon *s in shapeCollection.polygons) {
+            auto shapeLatLngs = coordinatesForShape(s);
+            latLngs.insert(latLngs.end(), shapeLatLngs.begin(), shapeLatLngs.end());
+        }
+    } else if ([shape isKindOfClass:[MGLMultiPolyline class]]) {
+        MGLMultiPolyline *shapeCollection = (MGLMultiPolyline *)shape;
+        for(MGLPolyline *s in shapeCollection.polylines) {
+            auto shapeLatLngs = coordinatesForShape(s);
+            latLngs.insert(latLngs.end(), shapeLatLngs.begin(), shapeLatLngs.end());
+        }
+    } else if ([shape isKindOfClass:[MGLShapeCollection class]]) {
+        MGLShapeCollection *shapeCollection = (MGLShapeCollection *)shape;
+        for(MGLShape *s in shapeCollection.shapes) {
+            auto shapeLatLngs = coordinatesForShape(s);
+            latLngs.insert(latLngs.end(), shapeLatLngs.begin(), shapeLatLngs.end());
```

Comment:
Consider having the mbgl::Map method take a Geometry; each MGLShape subclass already has an internal method for converting to a Geometry.